cargo-typify/src/main.rs

```diff
@@ -0,0 +1,32 @@
+use cargo_typify::{convert, CliArgs};
+use clap::Parser;
+
+use color_eyre::eyre::{Context, Result};
+
+#[derive(Parser)] // requires `derive` feature
+#[command(name = "cargo")]
+#[command(bin_name = "cargo")]
+enum CargoCli {
+    Typify(CliArgs),
+}
```

Comment:
This was added to make the cargo sub-command work nicely!

To run the cli locally you now need to do something like the following:

```
cargo run -- typify some_file.json
```

This style was taken from the Clap cookbook: https://docs.rs/clap/latest/clap/_derive/_cookbook/cargo_example_derive/index.html